sdk/src/main/java/com/stormpath/sdk/models/UserProfile.java

```diff
@@ -0,0 +1,65 @@
+package com.stormpath.sdk.models;
+
+import com.squareup.moshi.Json;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public class UserProfile implements Serializable {
+
+    @Json(name = "email")
+    private String email;
+
+    @Json(name = "givenName")
+    private String givenName;
+
+    @Json(name = "middleName")
+    private String middleName;
+
+    @Json(name = "surname")
+    private String surname;
+
+    @Json(name = "fullName")
+    private String fullName;
+
+    @Json(name = "status")
+    private String status;
+
+    @Json(name = "username")
+    private String username;
+
+    @Json(name = "customData")
+    private Map<String, String> customData;
```

Comment:
This needs to be a `Map<String, Object>`

custom data can be any structured json, so a `key` could have `value`  of `String` or `String[]` or `Object`, for instance.